activerecord/lib/active_record/counter_cache.rb

```diff
@@ -65,7 +65,7 @@ def reset_counters(id, *counters, touch: nil)
           updates.merge!(touch_updates)
         end
 
-        unscoped.where(primary_key => object.id).update_all(updates) if updates.any?
+        unscoped.where(primary_key => [object.id]).update_all(updates) if updates.any?
```

Comment:
In case if someone will be following changes history - this was pushed unintentionally and will be removed in https://github.com/rails/rails/pull/47221